app/controllers/reservations_controller.rb

```diff
@@ -138,6 +138,17 @@ def show
     @reservation  = Reservation.find(params[:id])
 
     raise ActiveRecord::RecordNotFound if @reservation != @order_detail.reservation
+
+    respond_to do |format|
+      format.html
+
+      format.ics do
+        calendar = ReservationCalendar.new(@reservation, request.host)
+        send_data(calendar.to_ical,
```

Comment:
I think this is mostly fine, though we may want to consider abstracting it up if we add this to other controllers. Look at what we did for CSVs in `config/initializers/csv_renderer.rb`. I'd be fine if you say, let's deal with that when we add it to more places.